scripts/ci/gitlab/pipeline/test.yml

```diff
@@ -72,3 +76,31 @@ test-deterministic-wasm:
     - .compiler-info
   script:
     - ./scripts/ci/gitlab/test_deterministic_wasm.sh
+
+cargo-fmt:
+  stage:                           test
+  variables:
+    RUSTY_CACHIER_TOOLCHAIN: nightly
+  extends:
+    - .docker-env
+    - .test-refs
+  script:
+    # - rusty-cachier snapshot create
+    - cargo +nightly fmt --all -- --check
+    # - rusty-cachier cache upload
+
+cargo-clippy:
+  stage:                           test
+  # this is an artificial job dependency, for pipeline optimization using GitLab's DAGs
+  needs:
+    - job:                         cargo-fmt
+      artifacts:                   false
+  variables:
+    RUSTY_CACHIER_TOOLCHAIN: nightly
+  extends:
+    - .docker-env
+    - .test-refs
+  script:
+    # - rusty-cachier snapshot create
+    - SKIP_WASM_BUILD=1 env -u RUSTFLAGS cargo +nightly clippy --all-targets
```

Comment:
Why unsetting `RUSTFLAGS`? rusty-cachier could be useful here.

Reply:
I just copied what substrate does https://github.com/paritytech/substrate/blob/master/scripts/ci/gitlab/pipeline/test.yml#L51

Reply:
Where do we get `rusty-cachier` from? I didn't see anything in the substrate repo. Google didn't help either.

Reply:
It is a project from @rcny

Reply:
We are going to add `rusty-cachier` to `cumulus` and `polkadot` but now it won't work if you enable it here. So the commented lines can be removed.

Reply:
Correct. Current references could be removed as we'll instrument the project with `rusty-cachier` by ourselves later.

> Where do we get `rusty-cachier` from?

From [here](https://gitlab.parity.io/parity/infrastructure/ci_cd/rusty-cachier/client). You can read more about the project [here](https://gitlab.parity.io/groups/parity/infrastructure/ci_cd/rusty-cachier/-/wikis/Architecture).